prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt

```diff
@@ -141,4 +155,112 @@ class DatabaseDeliveryAccess(
             query.fetchInto(DeliveryFacility::class.java)
         }
     }
+
+    /**
+     * Get a list of matching delivery facilities for a set of [reportIds]
+     * @todo further work on https://github.com/CDCgov/prime-reportstream/issues/9272
+     * @param reportIds list of reports to check facilities for
+     * @return List of matching delivery facilities
+     */
+    fun fetchBulkDeliveryFacilities(
+        receiver: String,
+        receivingOrgSvc: String?,
+        sortDir: SortDir,
+        sortColumns: List<BulkFacilitySortColumn>,
+        since: OffsetDateTime?,
+        until: OffsetDateTime?,
+        pageSize: Int,
+        pageNumber: Int
+    ): List<DeliveryFacility> {
+        val reportIds = db.transactReturning { txn ->
+            var filter = this.organizationFilter(receiver, receivingOrgSvc)
+            if (since != null) {
+                filter = filter.and(ACTION.CREATED_AT.ge(since))
+            }
+            if (until != null) {
+                filter = filter.and(ACTION.CREATED_AT.lt(until))
+            }
+
+            val query = DSL.using(txn)
+                // Note the report file and action tables have columns with the same name, so we must specify what we need.
+                .select(REPORT_FILE.REPORT_ID)
+                .from(
+                    ACTION.join(REPORT_FILE).on(
+                        REPORT_FILE.ACTION_ID.eq(ACTION.ACTION_ID)
+                    )
+                )
+                .where(filter)
+                .groupBy(REPORT_FILE.REPORT_ID)
+
+            query.fetchInto(ReportId::class.java)
+        }
+
+        return db.transactReturning { txn ->
+            var filter = COVID_RESULT_METADATA.REPORT_ID.isNotNull()
+            if (since != null) {
```

Comment:
Do we still need this filter since were doing a date check above? (There also is not an index on that column)

Reply:
True, we don't need that second filter!